src/front/wgsl/mod.rs

```diff
@@ -1232,12 +1232,11 @@ impl Parser {
         mut handle: Handle<crate::Expression>,
         allow_deref: bool,
     ) -> Result<Handle<crate::Expression>, Error<'a>> {
-        let mut needs_deref = match ctx.expressions[handle] {
-            crate::Expression::LocalVariable(_) | crate::Expression::GlobalVariable(_) => {
-                allow_deref
-            }
-            _ => false,
-        };
+        let mut needs_deref = allow_deref
```

Comment:
Is this really better than the old code?
I think we should only use macros where their benefit outweight the cost of having an extra layer of indirection. Macros slow down compilation, confuse IDEs, and generally shouldn't be used so sparingly.

Reply:
Okay, this is the case where I think `matches!` actually works. Putting `allow_deref` in the first arm of the `match` is a really obscure way of saying `&&`. And if you do use `&&`, then the `match` just has arms with `=> true` and `=> false`, which is classic `matches!`. Since this is a `std` macro, I think IDEs are going to pick up on it pretty quickly; `rust-analyzer`, at least, hasn't been having any problems with it.

But if you don't find that persuasive, it's not something I feel strongly about.

Reply:
I think I agree. It's not striking me as a case that would need to be an exception though if we try to avoid matches in general (I checked briefly and didn't find a clippy for this).